akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -256,24 +256,30 @@ abstract class ActorMaterializer extends Materializer with MaterializerLoggingPr
  */
 class MaterializationException(msg: String, cause: Throwable = null) extends RuntimeException(msg, cause)
 
+/**
+ * A base exception for abrupt stream termination.
+ * */
+sealed class AbruptStreamTerminationException(msg: String, cause: Throwable)
```

Comment:
Since sealed doesn't show up for Java users and this is shared API it also needs

```suggestion
 * A base exception for abrupt stream termination.
 * 
 * Not for user extension 
 */
@DoNotInherit
sealed class AbruptStreamTerminationException(msg: String, cause: Throwable)
```